pkg/controller/deployment/deployment_controller.go

```diff
@@ -242,6 +276,33 @@ func (d *DeploymentController) reconcileOldRCs(allRCs []*api.ReplicationControll
 	return true, err
 }
 
+func (d *DeploymentController) scaleDownOldRCs(oldRCs []*api.ReplicationController, deployment extensions.Deployment) (bool, error) {
+
+	scaled := false
+
+	for _, rc := range oldRCs {
+		// Scaling not required.
+		if rc.Spec.Replicas == 0 {
+			continue
+		}
+		scaled = true
+		_, err := d.scaleRCAndRecordEvent(rc, 0, deployment)
+		if err != nil {
+			return false, err
+		}
+	}
+	return scaled, nil
+}
+
+func (d *DeploymentController) scaleUpNewRC(newRC *api.ReplicationController, deployment extensions.Deployment) (bool, error) {
```

Comment:
Add a comment noting that this method is intended to be used with the recreate strategy.